src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -205,18 +197,17 @@ class ElementWidgetIcons {
             poskey = typekey->second.find(pos);
         }
 
-        if (icontype == Type::Normal)
+        if (icontype == ElementItem::GeometryState::Normal)
             return std::get<0>(poskey->second);
-        else if (icontype == Type::Construction)
+        else if (icontype == ElementItem::GeometryState::Construction)
             return std::get<1>(poskey->second);
-        else if (icontype == Type::External)
+        else if (icontype == ElementItem::GeometryState::External)
             return std::get<2>(poskey->second);
-        else //internal
+        else //internal alignment

```

Comment:
How about using a `switch` here ?

Reply:
@abdullahtahiriyo can you please just comment here? Is my last point about code itself. :)